src/System.IO.FileSystem/src/System/IO/FileStatus.Unix.cs

```diff
@@ -188,15 +187,14 @@ internal DateTimeOffset GetLastWriteTime(ReadOnlySpan<char> path, bool continueO
             return UnixTimeToDateTimeOffset(_fileStatus.MTime, _fileStatus.MTimeNsec);
         }
 
-        internal void SetLastWriteTime(string path, DateTimeOffset time)
-            => SetAccessWriteTimes(path, null, time.ToUnixTimeSeconds());
-
+        internal void SetLastWriteTime(string path, DateTimeOffset time) => SetAccessWriteTimes(path, time, isAccessTime: false);
+        
         private DateTimeOffset UnixTimeToDateTimeOffset(long seconds, long nanoseconds)
         {
             return DateTimeOffset.FromUnixTimeSeconds(seconds).AddTicks(nanoseconds / NanosecondsPerTick).ToLocalTime();
         }
 
-        private void SetAccessWriteTimes(string path, long? accessTime, long? writeTime)
+        private void SetAccessWriteTimes(string path, DateTimeOffset time, bool isAccessTime)
```

Comment:
Nit: shouldn't this method now be named SetAccessOrWriteTime?  It's not longer capable of setting both at the same time as it was previously.